### PR TITLE
2 returning learning path via api endpoint

### PR DESCRIPTION
**Pull request by MarcNor**
The learning path can now be retrieved via the API ndpoint "\learningPath" with the needed body parameters _studentId_ and _learningStyle_.

### CODE REVIEW

File: domain/tutoringModel/learning_path.py [hunk 1]
```diff
@@ -61,23 +61,33 @@ def calculate_sequence(self, learning_element_types, input_learning_style):
...
             learning_path, key=learning_path.get, reverse=True)
 
         for w in sorted_keys:
-            sort_learning_path[w] = learning_path[w]
-
+            sort_learning_path_dict[w] = learning_path[w]
+
+        print(sort_learning_path_dict)
+        for _ in range(len(learning_path)):
+            test = max(learning_path, key=learning_path.get)
```
Comment (yusme) [inline domain/tutoringModel/learning_path.py:76]: The variable "test" is unclear, it's a key or a learning_elements

File: entrypoints/flask_app.py [hunk 2]
```diff
@@ -1,11 +1,30 @@
...
 orm.start_mappers()
 
 
-@app.route("/")
-def func():
-    return jsonify(), 200
+@app.route("/learningPath")
+def get_learning_path():
+    if request.json is None or 'studentId' not in request.json:
+        return jsonify(), 404
```
Comment (yusme) [inline entrypoints/flask_app.py:14]: The number 404 ist unclear, it's a error_code or a status ?

File: entrypoints/flask_app.py [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
+                unit_of_work.SqlAlchemyUnitOfWork(),
+                request.json["learningStyle"]
+            )
+        else:
+            learning_path = services.get_learning_path(
+                unit_of_work.SqlAlchemyUnitOfWork()
+            )
+
+        if type(learning_path) is ValueError:
+            return jsonify({}), 400
```
Comment (yusme) [inline entrypoints/flask_app.py:27]: The number 400 ist unclear, it's a error_code or a status ?

File: entrypoints/flask_app.py [hunk 4]
```diff
@@ -1,11 +1,30 @@
...
+        else:
+            dict = {'learningPath': learning_path}
+            return jsonify(dict), 200
```
Comment (yusme) [inline entrypoints/flask_app.py:30]: The number 200 ist unclear, it's a error_code or a status ?

File: entrypoints/flask_app.py [hunk 5]
```diff
@@ -1,11 +1,30 @@
-from flask import Flask, jsonify
-
+from multiprocessing.sharedctypes import Value
+from flask import Flask, jsonify, request
+from domain.tutoringModel import learning_path
```
Comment (yusme) [inline entrypoints/flask_app.py:3]: This is not used anywhere

File: tests/integration/test_uow.py [hunk 6]
```diff
@@ -1,6 +1,5 @@
 from select import select
```
Comment (yusme) [inline tests/integration/test_uow.py:1]: This is not used anywhere

**Review by yusme — COMMENTED** [review]
.

File: domain/tutoringModel/learning_path.py [hunk 7]
```diff
@@ -61,23 +61,33 @@ def calculate_sequence(self, learning_element_types, input_learning_style):
         return sum
 
     def sort_learning_path(self, learning_path):
-        sort_learning_path = {}
+        print(learning_path)
```
[the remark this answers is not in the document]
Comment (MarcNor) [reply domain/tutoringModel/learning_path.py:64]: removed

File: domain/tutoringModel/learning_path.py [hunk 8]
```diff
@@ -61,23 +61,33 @@ def calculate_sequence(self, learning_element_types, input_learning_style):
...
+
         sorted_keys = sorted(
             learning_path, key=learning_path.get, reverse=True)
 
         for w in sorted_keys:
-            sort_learning_path[w] = learning_path[w]
-
+            sort_learning_path_dict[w] = learning_path[w]
+
+        print(sort_learning_path_dict)
```
[the remark this answers is not in the document]
Comment (MarcNor) [reply domain/tutoringModel/learning_path.py:74]: removed

File: domain/tutoringModel/learning_path.py [hunk 9]
```diff
@@ -61,23 +61,33 @@ def calculate_sequence(self, learning_element_types, input_learning_style):
...
+        for _ in range(len(learning_path)):
+            test = max(learning_path, key=learning_path.get)
+            sort_learning_path.append(test)
+            del learning_path[test]
+        print(','.join(sort_learning_path))
```
[the remark this answers is not in the document]
Comment (MarcNor) [reply domain/tutoringModel/learning_path.py:79]: removed

File: domain/tutoringModel/learning_path.py (hunk 1 above)
Comment (MarcNor) [reply domain/tutoringModel/learning_path.py:76]: renamed

File: entrypoints/flask_app.py (hunk 2 above)
Comment (MarcNor) [reply entrypoints/flask_app.py:14]: updated

File: entrypoints/flask_app.py (hunk 3 above)
Comment (MarcNor) [reply entrypoints/flask_app.py:27]: updated

File: entrypoints/flask_app.py (hunk 4 above)
Comment (MarcNor) [reply entrypoints/flask_app.py:30]: updated

File: entrypoints/flask_app.py (hunk 5 above)
Comment (MarcNor) [reply entrypoints/flask_app.py:3]: removed

File: tests/integration/test_uow.py (hunk 6 above)
Comment (MarcNor) [reply tests/integration/test_uow.py:1]: removed

**Review by yusme — APPROVED** [review]

**Pull request merged**